### PR TITLE
[Rated Disabilities] Prepping for rated disabilities LH migration PR

### PR DESCRIPTION
**Pull request by jerekshoe**
## Summary
Prepping for the PR to add the LH migration code to the application. Changes include:
* Moving MVI error alert into its' own component
* Relocating logic for displaying the MVI error alert to the `App` component (this way I don't have to duplicate this in the next PR)
* Using `va-link` in `components/TotalRatingStates.jsx` instead of `a` tag
* Updating unit tests to reflect changes made when moving the MVI error logic

## Related issue(s)
- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#63310

## Testing done
Updating tests to account for relocation of logic for displaying MVI errors from `RatedDisabilityView` to -> `App` component

## What areas of the site does it impact?
Rated Disabilities Application

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/rated-disabilities/components/TotalRatingStates.jsx [hunk 1]
```diff
@@ -30,13 +30,12 @@ export const missingTotalMessage = () => (
...
     </p>
-    <a
-      href="/disability/how-to-file-claim/"
-      className="usa-link vads-u-font-size--base"
-      aria-label="Learn how to file a claim for disability compensation"
+    <va-link
+      href="/disability/how-to-file-claim"
+      text="Learn how to file a claim for disability compensation"
     >
       Learn how to file a claim for disability compensation
```
Comment (Mottie) [inline src/applications/rated-disabilities/components/TotalRatingStates.jsx:37]: Remove this?

File: src/applications/rated-disabilities/tests/components/RatedDisabilityList.unit.spec.jsx [hunk 2]
```diff
@@ -9,7 +9,7 @@ describe('<RatedDisabilityList/>', () => {
   const ratedDisabilities = {
     ratedDisabilities: [
       {
-        decisionCode: 'SVCCONNECTED',
+        decisionCode: 'SVCCONNCTED',
```
Comment (Mottie) [inline src/applications/rated-disabilities/tests/components/RatedDisabilityList.unit.spec.jsx:12]: Weird that I see this spelled both ways in the code base...

**Review by Mottie — DISMISSED** [review]

File: src/applications/rated-disabilities/tests/components/RatedDisabilityList.unit.spec.jsx (hunk 2 above)
Comment (jerekshoe) [reply src/applications/rated-disabilities/tests/components/RatedDisabilityList.unit.spec.jsx:12]: I looked at what EVSS actually sends back in staging and the vets-api code, but probably needs to be updated elsewhere as well

File: src/applications/rated-disabilities/components/TotalRatingStates.jsx (hunk 1 above)
Comment (jerekshoe) [reply src/applications/rated-disabilities/components/TotalRatingStates.jsx:37]: Ah yeah, copy-paste error

Comment (jerekshoe) [reply src/applications/rated-disabilities/components/TotalRatingStates.jsx]: Fixed

File: src/applications/rated-disabilities/tests/components/RatedDisabilityList.unit.spec.jsx (hunk 2 above)
Comment (jerekshoe) [reply src/applications/rated-disabilities/tests/components/RatedDisabilityList.unit.spec.jsx:12]: I updated the other files that use the incorrect value

**Review by Mottie — APPROVED** [review]

**Pull request merged**